File: 0100. Same Tree/solution_1.java
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     int val;
 *     TreeNode left;
 *     TreeNode right;
 *     TreeNode() {}
 *     TreeNode(int val) { this.val = val; }
 *     TreeNode(int val, TreeNode left, TreeNode right) {
 *         this.val = val;
 *         this.left = left;
 *         this.right = right;
 *     }
 * }
 */

class Solution 
{
    public void WalkTree(TreeNode p, TreeNode q, boolean[] same)
    {
        if (p != null && q != null)
        {
            if (p.val != q.val)
            {
                same[0] = false;
                return;
            }
            
            WalkTree(p.left, q.left, same);
            WalkTree(p.right, q.right, same);
        }
        else if (p != null && q == null || p == null && q != null)
        {
            same[0] = false;
            return;
        }
    }
    
    public boolean isSameTree(TreeNode p, TreeNode q) 
    {
        boolean[] same = {true};
        WalkTree(p, q, same);
        return same[0];
    }
}

File: leetcode/100-999/100. Same Tree/solution_1.js
/**
 * Definition for a binary tree node.
 * function TreeNode(val, left, right) {
 *     this.val = (val===undefined ? 0 : val)
 *     this.left = (left===undefined ? null : left)
 *     this.right = (right===undefined ? null : right)
 * }
 */

/** 
 * @param {TreeNode} p
 * @param {TreeNode} q
 * @param {object} same
 * @return {void}
 * **/

var walkTree = function(p, q, same)
{
    if (p && q)
    {
        if (p.val != q.val)
        {
            same.value = false
            return
        }

        walkTree(p.left, q.left, same)
        walkTree(p.right, q.right, same)
    }
    else if ((p && q == null) || (p == null && q))
    {
        same.value = false
        return
    }
}

/**
 * @param {TreeNode} p
 * @param {TreeNode} q
 * @return {boolean}
 */

var isSameTree = function(p, q) {
    const same = { value: true }
    walkTree(p, q, same)
    return same.value
};